Test Failure: JDBCLoadFromAppTest.testLoginModuleFromEJBInWebApp access denied modifyPrivateCredentials

Test Failure (20200303-0657): com.ibm.ws.jdbc.fat.loadfromapp.JDBCLoadFromAppTest.testLoginModuleFromEJBInWebApp

```
junit.framework.AssertionFailedError: 2020-03-03-16:36:21:843 The response did not contain "[SUCCESS]".  Full output is:"
ERROR: Caught exception attempting to call test method testLoginModuleFromEJBInWebApp on servlet web.derby.DerbyLoadFromAppServlet
javax.ejb.EJBException: See nested exception; nested exception is: java.lang.RuntimeException: java.sql.SQLException: javax.security.auth.login.LoginException: Access denied ("javax.security.auth.AuthPermission" "modifyPrivateCredentials")
	at com.ibm.ejs.container.util.ExceptionUtil.EJBException(ExceptionUtil.java:399)
	at com.ibm.ejs.container.util.ExceptionUtil.EJBException(ExceptionUtil.java:285)
	at com.ibm.ejs.container.BusinessExceptionMappingStrategy.mapException(BusinessExceptionMappingStrategy.java:347)
	at com.ibm.ejs.container.BusinessExceptionMappingStrategy.setUncheckedException(BusinessExceptionMappingStrategy.java:519)
	at com.ibm.ejs.container.EJSDeployedSupport.setUncheckedLocalException(EJSDeployedSupport.java:449)
	at com.ibm.ejs.container.deploy.java.util.concurrent.EJSLocal0SLBeanInWebApp_d1066e83.execute(EJSLocal0SLBeanInWebApp_d1066e83.java)
	at web.derby.DerbyLoadFromAppServlet.testLoginModuleFromEJBInWebApp(DerbyLoadFromAppServlet.java:84)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at componenttest.app.FATServlet.doGet(FATServlet.java:71)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:686)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:791)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1230)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:729)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:426)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.invokeTarget(WebAppFilterChain.java:182)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:93)
	at com.ibm.ws.security.jaspi.JaspiServletFilter.doFilter(JaspiServletFilter.java:56)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:201)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:90)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.doFilter(WebAppFilterManager.java:1001)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1139)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1010)
	at com.ibm.ws.webcontainer.servlet.CacheServletWrapper.handleRequest(CacheServletWrapper.java:75)
	at com.ibm.ws.webcontainer40.servlet.CacheServletWrapper40.handleRequest(CacheServletWrapper40.java:83)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:938)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:279)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1134)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:415)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:374)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:548)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:482)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:347)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:318)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:167)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:75)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:504)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:574)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:958)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1047)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:825)
Caused by: java.lang.RuntimeException: java.sql.SQLException: javax.security.auth.login.LoginException: Access denied ("javax.security.auth.AuthPermission" "modifyPrivateCredentials")
	at web.derby.DerbyLoadFromAppServlet.lambda$testLoginModuleFromEJBInWebApp$0(DerbyLoadFromAppServlet.java:95)
	at web.derby.DerbyLoadFromAppServlet$$Lambda$267.000000001003A0C0.run(Unknown Source)
	at web.derby.BeanInWebApp.execute(BeanInWebApp.java:26)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at com.ibm.ejs.container.EJSContainer.invokeProceed(EJSContainer.java:4886)
	at com.ibm.ejs.container.interceptors.InvocationContextImpl.proceed(InvocationContextImpl.java:592)
	at com.ibm.ws.cdi.ejb.impl.InterceptorChain.proceed(InterceptorChain.java:119)
	at com.ibm.ws.cdi.ejb.impl.EJBCDIInterceptorWrapper.invokeInterceptors(EJBCDIInterceptorWrapper.java:131)
	at com.ibm.ws.cdi.ejb.impl.EJBCDIInterceptorWrapper.aroundInvoke(EJBCDIInterceptorWrapper.java:54)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at com.ibm.ejs.container.interceptors.InterceptorProxy.invokeInterceptor(InterceptorProxy.java:189)
	at com.ibm.ejs.container.interceptors.InvocationContextImpl.proceed(InvocationContextImpl.java:577)
	at org.jboss.weld.module.ejb.AbstractEJBRequestScopeActivationInterceptor.aroundInvoke(AbstractEJBRequestScopeActivationInterceptor.java:72)
	at com.ibm.ws.cdi.ejb.interceptor.WeldSessionBeanInterceptorWrapper.aroundInvoke(WeldSessionBeanInterceptorWrapper.java:58)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at com.ibm.ejs.container.interceptors.InterceptorProxy.invokeInterceptor(InterceptorProxy.java:189)
	at com.ibm.ejs.container.interceptors.InvocationContextImpl.proceed(InvocationContextImpl.java:577)
	at com.ibm.ejs.container.interceptors.InvocationContextImpl.doAroundInterceptor(InvocationContextImpl.java:298)
	at com.ibm.ejs.container.interceptors.InvocationContextImpl.doAroundInvoke(InvocationContextImpl.java:267)
	at com.ibm.ejs.container.EJSContainer.invoke(EJSContainer.java:4788)
Caused by: java.sql.SQLException: javax.security.auth.login.LoginException: Access denied ("javax.security.auth.AuthPermission" "modifyPrivateCredentials")
	at com.ibm.ws.rsadapter.AdapterUtil.toSQLException(AdapterUtil.java:841)
	at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:147)
	at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:112)
	at web.derby.DerbyLoadFromAppServlet.lambda$testLoginModuleFromEJBInWebApp$0(DerbyLoadFromAppServlet.java:87)
Caused by: javax.resource.ResourceException: javax.security.auth.login.LoginException: Access denied ("javax.security.auth.AuthPermission" "modifyPrivateCredentials")
	at com.ibm.ejs.j2c.ConnectionManager.getFinalSubject(ConnectionManager.java:1659)
	at com.ibm.ejs.j2c.ConnectionManager.allocateConnection(ConnectionManager.java:297)
	at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:138)
Caused by: javax.security.auth.login.LoginException: Access denied ("javax.security.auth.AuthPermission" "modifyPrivateCredentials")
	at web.derby.LoadFromWebAppLoginModule.login(LoadFromWebAppLoginModule.java:57)
	at com.ibm.ws.security.jaas.common.modules.WSLoginModuleProxy.login(WSLoginModuleProxy.java:141)
	at com.ibm.ws.kernel.boot.security.LoginModuleProxy.login(LoginModuleProxy.java:51)
	at java.base/javax.security.auth.login.LoginContext.invoke(LoginContext.java:726)
	at java.base/javax.security.auth.login.LoginContext$4.run(LoginContext.java:665)
	at java.base/javax.security.auth.login.LoginContext$4.run(LoginContext.java:663)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:739)
	at java.base/javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:663)
	at java.base/javax.security.auth.login.LoginContext.login(LoginContext.java:574)
	at com.ibm.ws.security.jca.internal.AuthDataServiceImpl.createSubjectUsingJAAS(AuthDataServiceImpl.java:122)
	at com.ibm.ws.security.jca.internal.AuthDataServiceImpl.getSubject(AuthDataServiceImpl.java:112)
	at com.ibm.ejs.j2c.ConnectionManager$1.run(ConnectionManager.java:1654)
	at com.ibm.ejs.j2c.ConnectionManager$1.run(ConnectionManager.java:1651)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:703)
	at com.ibm.ejs.j2c.ConnectionManager.getFinalSubject(ConnectionManager.java:1651)
Caused by: java.security.AccessControlException: Access denied ("javax.security.auth.AuthPermission" "modifyPrivateCredentials")
	at java.base/java.security.AccessController.throwACE(AccessController.java:176)
	at java.base/java.security.AccessController.checkPermissionHelper(AccessController.java:237)
	at java.base/java.security.AccessController.checkPermission(AccessController.java:373)
	at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:322)
	at java.base/javax.security.auth.Subject$SecureSet.add(Subject.java:1145)
	at java.base/java.util.Collections$SynchronizedCollection.add(Collections.java:2038)
	at web.derby.LoadFromWebAppLoginModule$1.run(LoadFromWebAppLoginModule.java:52)
	at web.derby.LoadFromWebAppLoginModule$1.run(LoadFromWebAppLoginModule.java:47)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:647)
	at web.derby.LoadFromWebAppLoginModule.login(LoadFromWebAppLoginModule.java:47)
". expected:<0> but was:<1>
	at componenttest.topology.utils.HttpUtils.findStringInHttpConnection(HttpUtils.java:539)
	at componenttest.topology.utils.HttpUtils.findStringInHttpConnection(HttpUtils.java:497)
	at componenttest.topology.utils.HttpUtils.findStringInUrl(HttpUtils.java:474)
	at componenttest.topology.utils.HttpUtils.findStringInReadyUrl(HttpUtils.java:446)
	at componenttest.topology.utils.HttpUtils.findStringInReadyUrl(HttpUtils.java:416)
	at componenttest.topology.utils.FATServletClient.runTest(FATServletClient.java:83)
	at componenttest.custom.junit.runner.SyntheticServletTest.invokeExplosively(SyntheticServletTest.java:40)
	at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:193)
	at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:314)
	at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:167)
```

It fails because one of the test applications which includes login modules lacks modifyPrivateCredentials permission, which is required by the login module.

--- a/dev/com.ibm.ws.jdbc_fat_loadfromapp/publish/servers/com.ibm.ws.jdbc.fat.loadfromapp/server.xml
+++ b/dev/com.ibm.ws.jdbc_fat_loadfromapp/publish/servers/com.ibm.ws.jdbc.fat.loadfromapp/server.xml
@@ -110,6 +110,7 @@
     <javaPermission codeBase="${server.config.dir}apps/otherApp.ear" className="java.lang.RuntimePermission" name="getClassLoader"/>
 
     <!-- permissions for login modules -->
+    <javaPermission codebase="${server.config.dir}/apps/derbyApp.war" className="javax.security.auth.AuthPermission" name="modifyPrivateCredentials"/>
     <javaPermission codebase="${server.config.dir}/apps/otherApp.ear" className="javax.security.auth.AuthPermission" name="modifyPrivateCredentials"/>
 
     <variable name="onError" value="FAIL"/>

